.github/workflows/docker_simulate-sce.yml

```diff
@@ -23,59 +19,40 @@ on:
       - analyses/simulate-sce/.dockerignore
       - analyses/simulate-sce/renv.lock
       - analyses/simulate-sce/conda-lock.yml
-
-permissions:
-  id-token: write # This is required for requesting the JWT
-  contents: read # This is required for actions/checkout
+  workflow_dispatch:
+    inputs:
+      push-ecr:
+        description: "Push to AWS ECR"
+        type: boolean
+        required: false
+  workflow_call:
+    inputs:
+      push-ecr:
+        description: "Push to AWS ECR"
+        type: boolean
 
 jobs:
-  build:
-    name: Build Docker Image
+  test-build:
+    name: Test Build Docker Image
+    if: github.event_name == 'pull_request' || (contains(github.event_name, 'workflow_') && !inputs.push-ecr)
```

Comment:
Do I have the right idea: If you're manually calling it or reusing a workflow and push is false, skip straight to `build-push`

Reply:
If you are manual or reuse and `push-ecr` is false, we run this, if `push-ecr` is true, we skip to `build-push`. (I made the variable different because `push` is overloaded)